deploy/chart/values.yaml

```diff
@@ -0,0 +1,70 @@
+# Default values for openshift-routes.
+
+replicas: 1
+logLevel: 5
+nameOverride: ""
+fullnameOverride: ""
+
+image:
+  # https://github.com/cert-manager/openshift-routes/pkgs/container/cert-manager-openshift-routes
+  registry: ghcr.io
+  repository: cert-manager/cert-manager-openshift-routes
```

Comment:
**Suggestion:** could we use `repository` to mean the full URL (registry domain + repository path) like we do in trust-manager and cert-manager?



In cert-manager for example: [`values.yaml`](https://github.com/cert-manager/cert-manager/blob/df4d15ce4ad6b80c1b53ed5908290e3f835c49b5/deploy/charts/cert-manager/values.yaml#L78C1-L92C1)

```yaml
image:
  repository: quay.io/jetstack/cert-manager-controller
  # You can manage a registry with
  # registry: quay.io
  # repository: jetstack/cert-manager-controller

  # Override the image tag to deploy by setting this variable.
  # If no value is set, the chart's appVersion will be used.
  # tag: canary

  # Setting a digest will override any tag
  # digest: sha256:0e072dddd1f7f8fc8909a2ca6f65e76c5f0d2fcfb8be47935ae3457e8bbceb20
  pullPolicy: IfNotPresent
```

Same with trust-manager: [`values.yaml`](https://github.com/cert-manager/trust-manager/blob/89f68eefc2c0d77d1cd82d95c824af6c8d284f4d/deploy/charts/trust-manager/values.yaml#L7C1-L22C27).

The trick is to add this bit of template to [`_helpers.tpl`](https://github.com/cert-manager/trust-manager/blob/89f68eefc2c0d77d1cd82d95c824af6c8d284f4d/deploy/charts/trust-manager/templates/_helpers.tpl#L28C1-L41C11):

```yaml
{{/*
Util function for generating the image URL based on the provided options.
IMPORTANT: This function is standarized across all charts in the cert-manager GH organization.
Any changes to this function should also be made in cert-manager, trust-manager, approver-policy, ...
See https://github.com/cert-manager/cert-manager/issues/6329 for a list of linked PRs.
*/}}
{{- define "image" -}}
{{- $defaultTag := index . 1 -}}
{{- with index . 0 -}}
{{- if .registry -}}{{ printf "%s/%s" .registry .repository }}{{- else -}}{{- .repository -}}{{- end -}}
{{- if .digest -}}{{ printf "@%s" .digest }}{{- else -}}{{ printf ":%s" (default $defaultTag .tag) }}{{- end -}}
{{- end }}
{{- end }}
```

WDYT?

Reply:
Hi, thanks for taking a look at the PR.

My idea for splitting the registry and the repository was to easily accommodate alternative registries or mirrors, such as `my-registry.example.com/quay.io/jetstack/cert-manager-controller`.

I totally that you want to keep the "common" values standardized across the various cert-manager repos, therefore I'll update this PR accordingly.

Reply:
I didn’t realize that folks were keeping the repository name as the « path » to the image.

note that it is still possible to do just that:

registry: my-registry.io
repository: quay.io/…
or repository: jetstack/cert-manager…

Does that match your expectation?